packages/salesforce-adapter/test/filters/workflow_field_update.test.ts

```diff
@@ -0,0 +1,118 @@
+  CUSTOM_OBJECT,
+  METADATA_TYPE,
+  WORKFLOW_FIELD_UPDATE_METADATA_TYPE,
+} from '../../src/constants'
+
+describe('WorklowFieldUpdate filter', () => {
+  const { client } = mockClient()
+
+  const filter = filterCreator({ client, config: {} }) as FilterWith<'onFetch'>
+
+  const workflowFieldUpdateInstanceName = 'update_user_signature'
+  const parentName = 'CustomUser'
+
+  const customParentObject = new ObjectType({
+    elemID: new ElemID(SALESFORCE, parentName),
+    annotations: { [METADATA_TYPE]: CUSTOM_OBJECT },
+  })
+
+  const generateWorkFlowFieldUpdateInstance = (
+    workflowParentName = parentName,
+    referencedFieldName = 'signature',
+  ): InstanceElement => {
+    const workflowFieldUpdateObjectType = new ObjectType({ elemID: WORKFLOW_FIELD_UPDATE_TYPE_ID })
+    const instanceName = `${workflowParentName}_${workflowFieldUpdateInstanceName}`
+    return new InstanceElement(
+      instanceName,
+      workflowFieldUpdateObjectType,
+      {
+        [INSTANCE_FULL_NAME_FIELD]: `${workflowParentName}${API_NAME_SEPERATOR}{workflowFieldUpdateInstanceName}`,
+        [WORKFLOW_FIELD_UPDATE_FIELD_FIELD]: referencedFieldName,
+        description: 'description',
+        name: `update ${referencedFieldName}`,
+        formula: `new ${referencedFieldName} 1`,
+        notifyAssignee: false,
+        operation: 'Formula',
+        protected: false,
+      },
+      [SALESFORCE, OBJECTS_PATH, WORKFLOW_FIELD_UPDATE_METADATA_TYPE, instanceName]
+    )
+  }
+
+  describe('on fetch', () => {
+    let workflowFieldUpdateType: ObjectType
+    let instance: InstanceElement
+    let elements: Element[]
+    let standardFieldObj: ObjectType
+    let unknownFieldInstance: InstanceElement
+    let unknownParentInstance: InstanceElement
+
+    beforeAll(async () => {
+      instance = generateWorkFlowFieldUpdateInstance()
+      workflowFieldUpdateType = instance.type
+
+      const testSobjPath = [SALESFORCE, OBJECTS_PATH, 'test']
+      const testSObj = customParentObject.clone()
+      testSObj.annotate({ [API_NAME]: parentName })
+      testSObj.path = testSobjPath
+
+      standardFieldObj = new ObjectType({
+        elemID: testSObj.elemID,
+        path: [SALESFORCE],
+        fields: { signature: { type: BuiltinTypes.STRING, annotations: { apiName: 'signature' } } },
+      })
+
+      unknownFieldInstance = generateWorkFlowFieldUpdateInstance(parentName, 'unknownField')
+      unknownParentInstance = generateWorkFlowFieldUpdateInstance('UnknownParent')
+
+      elements = [
+        testSObj, standardFieldObj, workflowFieldUpdateType, instance,
+        unknownFieldInstance, unknownParentInstance,
+      ]
+      await filter.onFetch(elements)
+    })
+
+    it('should have reference from field to parent field', () => {
+      const field = (
+        instance.value as PotentialFieldReferenceInstance
+      ).field as ReferenceExpression
+      expect(field.elemId).toEqual(standardFieldObj.fields.signature.elemID)
```

Comment:
fairly sure you don't need any of these casts and this can just be:
```suggestion
      expect(instance.value?.field).toBeInstanceOf(ReferenceExpression)
      expect(instance.value?.field?.elemId).toEqual(standardFieldObj.fields.signature.elemID)
```
(the first `expect` is really needed, I just added it so that if the test fails we'd get a more clear error)

same applies to other tests